zerver/lib/notification_data.py

```diff
@@ -352,4 +358,10 @@ def get_mentioned_user_group_name(
             smallest_user_group_size = current_user_group_size
             smallest_user_group_name = current_user_group.name
 
-    return smallest_user_group_name
+    if smallest_user_group_name is None:
+        return None
+
+    return MentionedUserGroup(
+        name=smallest_user_group_name,
+        members_count=int(smallest_user_group_size),  # for mypy
+    )
```

Comment:
I think you could make this cleaner if you just did

```
current_user_group = UserGroup.objects.get(id=user_group_id, realm=user_profile.realm)
current_mention = MentionedUserGroup(
    id=current_user_group.id,
    name=current_user_group.name, ...
)
if current_user_group.members_count < smallest_user_group_Size:
     smallest_mention = current_mention
```

that'll make mypy happier and also feel a bit cleaner.